src/viking.js

```diff
@@ -1,11 +1,84 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+  attack() {
+    return this.strength;
+  }
+  receiveDamage(damage) {
+    this.health -= damage;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+  receiveDamage(damage) {
+    this.health -= damage;
+    if (this.health <= 0) return `${this.name} has died in act of combat`;
+    return `${this.name} has received ${damage} points of damage`;
+  }
+  battleCry() {
+    return "Odin Owns You All!";
+  }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  constructor(health, strength) {
+    super(health, strength);
+  }
+  receiveDamage(damage) {
+    this.health -= damage;
+    if (this.health <= 0) return `A Saxon has died in combat`;
+    return `A Saxon has received ${damage} points of damage`;
+  }
+}
 
 // War
-class War {}
+class War {
+  constructor() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+  }
+  addViking(Viking) {
+    this.vikingArmy.push(Viking);
+  }
+  addSaxon(Saxon) {
+    this.saxonArmy.push(Saxon);
+  }
+  vikingAttack() {
+    let randomSaxon =
+      this.saxonArmy[Math.floor(Math.random() * this.saxonArmy.length)];
+    let randomViking =
+      this.vikingArmy[Math.floor(Math.random() * this.saxonArmy.length)];
+    let deadSaxons = randomSaxon.receiveDamage(randomViking.attack());
+    if (randomSaxon.health <= 0) {
+      this.saxonArmy.splice(Math.floor(Math.random() * this.saxonArmy.length),1)
```

Comment:
You are removing a random saxon from the `saxonArmy` array, and writing the logic again, when you can remove the exact random saxon that you previously randomized its index. That `randomSaxon` variable, already has the index of your dead saxon, so, you should use that index to remove from the array.